helpdesk_related_users/views/helpdesk_ticket_related.xml

```diff
@@ -0,0 +1,60 @@
+  <odoo>
+  <record id="helpdesk_ticket_search_inherit_related" model="ir.ui.view">
+    <field name="name">helpdesk.ticket.search.my.related in (helpdesk_related_users)</field>
+    <field name="model">helpdesk.ticket</field>
+    <field name="inherit_id" ref="helpdesk_mgmt.helpdesk_ticket_view_search"/>
+    <field name="arch" type="xml">
+      <xpath expr="//search" position="inside">
+        <filter string="My Related" name="my_related"
+                domain="[('related_user_ids', 'in', [uid])]"/>
+      </xpath>
+    </field>
+  </record>
+
+  <record id="helpdesk_ticket_form_inherit_related" model="ir.ui.view">
+    <field name="name">helpdesk.ticket.view.form.related.users in (helpdesk_related_users)</field>
+    <field name="model">helpdesk.ticket</field>
+    <field name="inherit_id" ref="helpdesk_mgmt.ticket_view_form"/>
+    <field name="arch" type="xml">
+      <xpath expr="//form/sheet/notebook" position="inside">
+        <page string="Collaborators">
+          <group>
+            <field name="related_user_ids"
+                   widget="many2many_avatar_user"
+                   domain="team_id and [('share','=',False), ('id','in', user_ids)] or [('id','=',False)]"
+                   options="{'no_create_edit': True}"
+                   context="{'default_share': False}"/>
+          </group>
+        </page>
+      </xpath>
+    </field>
+  </record>
+
+  <record id="helpdesk_ticket_tree_inherit_related" model="ir.ui.view">
+    <field name="name">helpdesk.ticket.view.tree.related.users in (helpdesk_related_users)</field>
+    <field name="model">helpdesk.ticket</field>
+    <field name="inherit_id" ref="helpdesk_mgmt.ticket_view_tree"/>
+    <field name="arch" type="xml">
+      <xpath expr="//tree/field[@name='user_id']" position="after">
+        <field name="related_user_ids"
+               widget="many2many_avatar_user"
+               optional="show"/>
```

Comment:
As an added field to a tree view with already a lot existing field, it's better to make it hidden as default